### PR TITLE
dai: rename soundwire to alh, add it to dai list and initialize it

### PR DESCRIPTION
**Pull request by slawblauciak**
Signed-off-by: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>

### CODE REVIEW

File: src/platform/icelake/include/platform/dai.h [hunk 1]
```diff
@@ -55,6 +55,11 @@
 /** \brief Number of HD/A Link Inputs */
 #define DAI_NUM_HDA_IN		7
 
+/* SoundWire */
+
+/** \brief Number of SoundWire multi-gateways */
+#define DAI_NUM_SOUNDWIRE_MULTI_GTW	16
```
Comment (tlauda) [inline src/platform/icelake/include/platform/dai.h]: I'm not sure how it is done for other interfaces, but shouldn't we define it for every cAVS platform? Right now enabling it on e.g. cnl will cause build error.

Comment (plbossart) [reply src/platform/icelake/include/platform/dai.h]: hmm, what is a multi-gateway? This concept is not part of the hardware. The hardware is seen by the firmware as just a set of GP-DMA channels going into specific FIFOs

Comment (plbossart) [inline src/platform/icelake/include/platform/dai.h]: Also what is this value of 16 all about? Is this 4*4 for PCM?
You can have 6 PDIs per link (2 for bulk and 4 for PCM), plus 4 for PDM in. This should really be based on concepts such as BSS, ISS, OSS for PCM and PDM.

Comment (slawblauciak) [reply src/platform/icelake/include/platform/dai.h]: @tlauda I knew I'd forget about it again...
@plbossart The plan is to aggregate multiple streams into multi-gateways like this.

**Comment by lgirdwood** [comment]
@slawblauciak also please explain ALH in the comments.

File: src/platform/icelake/include/platform/dai.h (hunk 1 above)
Comment (plbossart) [reply src/platform/icelake/include/platform/dai.h]: @slawblauciak the firmware is not aware of any aggregation. It does see a set of pipelines and DMA transfers going to different PDIs via the Audio Link Hub. The aggregation is controlled by the driver which enables the PDIs/Ports at the same time.

File: src/include/sof/soundwire.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (jajanusz) [inline src/include/sof/soundwire.h]: Please use spdx

File: src/platform/icelake/include/platform/dai.h (hunk 1 above)
Comment (slawblauciak) [reply src/platform/icelake/include/platform/dai.h]: @plbossart We have 4 segments and each has 4 bi-directional ports. Technically we have more ports available (there are also input only and output only ports), we're just ignoring them for now.

**Comment by slawblauciak** [comment]
Changes applied.

**Comment by jajanusz** [comment]
@slawblauciak I think there are too many things in commit message, can you split it into multiple commits?

**Comment by slawblauciak** [comment]
Done

**Comment by plbossart** [comment]
> ALH is an Intel specific name for the IP and is not generic. This should be kept as "soundwire" for all external interfaces. Documentation should use explain that soundwire is ALH.

so what? SSP is an Intel-specific name for an IP that caters to I2S and TDM modes.
It's perfectly fine to use ALH and mention that it *can* be connected to SoundWire but *may* be used in combination with other interfaces.
So no, ALH is the right way to represent hardware capabilities.

**Comment by slawblauciak** [comment]
Yes, while ALH is not generic, having it under the SoundWire name would be misleading for two reasons.
Firstly, the FW is not supposed to have any actual SoundWire-specific code.
Secondly, the module you'd call "SoundWire" would in actuality be used to connect to a variety of different interfaces behind the ALH, not just SoundWire.

+ @mmaka1

**Comment by lgirdwood** [comment]
@pl

> > ALH is an Intel specific name for the IP and is not generic. This should be kept as "soundwire" for all external interfaces. Documentation should use explain that soundwire is ALH.
> 
> so what? SSP is an Intel-specific name for an IP that caters to I2S and TDM modes.
> It's perfectly fine to use ALH and mention that it _can_ be connected to SoundWire but _may_ be used in combination with other interfaces.
> So no, ALH is the right way to represent hardware capabilities.

Really ? Are we going to have a vendor name enum (for DAI types) for every soundwire implementation from each vendor ? We don't have this for HDA. This differentiation just add unnecessary code.

**Comment by slawblauciak** [comment]
Closing due to further changes in the scope of the PR.
Please further discuss here: https://github.com/thesofproject/sof/pull/1638

**Pull request closed without merging**